src/auth/ha-auth-external.ts

```diff
@@ -32,9 +32,13 @@ export class HaAuthExternal extends HaForm {
 
   protected firstUpdated(changedProps) {
     super.firstUpdated(changedProps);
-    window.open(this.step.url);
+    const source = window.open(this.step.url);
     window.addEventListener("message", async (message: MessageEvent) => {
-      if (message.data.type === "externalCallback") {
+      if (
+        message.origin === window.location.origin &&
```

Comment:
The message is sent from the external site right? So the origin would something like `this.step.url`?

Reply:
It appears to always be the home assistant url e.g. `http://localhost:8123` when i'm testing, and not as the external IDP.